README.md

```diff
@@ -0,0 +1,343 @@
+retriever = DensePassageRetriever(document_store=document_store,
+                                  query_embedding_model="facebook/dpr-question_encoder-single-nq-base",
+                                  passage_embedding_model="facebook/dpr-ctx_encoder-single-nq-base",
+                                  use_gpu=True,
+                                  batch_size=16,
+                                  embed_title=True)
+retriever.retrieve(query="Why did the revenue increase?")
+# returns: [Document, Document]
+```
+
+-> See [docs](https://haystack.deepset.ai/docs/latest/retrievermd) for details
+
+
+### 5) Readers
+
+**What**  
+Neural networks (i.e. mostly Transformer-based) that read through texts in detail to find an answer. Use diverse models like BERT, RoBERTa or
+XLNet trained via [FARM](https://github.com/deepset-ai/FARM) or on SQuAD-like datasets. The Reader takes multiple passages of text as input
+and returns top-n answers with corresponding confidence scores. Both readers can load either a local model or any public model from [Hugging
+Face's model hub](https://huggingface.co/models)
+
+**Available Options**   
+- FARMReader: Reader based on [FARM](https://github.com/deepset-ai/FARM) incl. extensive configuration options and speed optimizations
+- TransformersReader: Reader based on the `pipeline` class of HuggingFace's [Transformers](https://github.com/huggingface/transformers).  
+**Both** Readers can load models directly from HuggingFace's model hub.
+
+**Example**  
+
+```python
+reader = FARMReader(model_name_or_path="deepset/roberta-base-squad2",
+                use_gpu=False, no_ans_boost=-10, context_window_size=500,
+                top_k_per_candidate=3, top_k_per_sample=1,
+                num_processes=8, max_seq_len=256, doc_stride=128)
+
+# Optional: Training & eval
+reader.train(...)
+reader.eval(...)
+
+# Predict
+reader.predict(question="Who is the father of Arya Starck?", documents=documents, top_k=3)
+```
+-> See [docs](https://haystack.deepset.ai/docs/latest/readermd) for details
+
+### 6) REST API
+**What**  
+A simple REST API based on [FastAPI](https://fastapi.tiangolo.com/) is provided to:
+
+-   search answers in texts ([extractive QA](https://github.com/deepset-ai/haystack/blob/master/rest_api/controller/search.py))
+-   search answers by comparing user question to existing questions
+    ([FAQ-style QA](https://github.com/deepset-ai/haystack/blob/master/rest_api/controller/search.py))
+-   collect & export user feedback on answers to gain domain-specific
+    training data
+    ([feedback](https://github.com/deepset-ai/haystack/blob/master/rest_api/controller/feedback.py))
+-   allow basic monitoring of requests (currently via APM in Kibana)
+
+**Example**  
+To serve the API, adjust the values in `rest_api/config.py` and run:
+
+    gunicorn rest_api.application:app -b 0.0.0.0:8000 -k uvicorn.workers.UvicornWorker -t 300
+
+You will find the Swagger API documentation at
+<http://127.0.0.1:8000/docs>
```

Comment:
This link didn't work for me. Is it because I didn't manage to serve the API?

Reply:
Yes, you need to start the API first as the link is pointing to your localhost